lib/stdlib/doc/src/proc_lib.xml

```diff
@@ -207,6 +225,65 @@ init(Parent) ->
     end,
     loop().
 
+...</code>
+      </desc>
+    </func>
+
+    <func>
+      <name since="OTP-26.0">init_fail(Ret, Exception) -> no_return()</name>
+      <name since="OTP-26.0">init_fail(Parent, Ret, Exception) -> no_return()</name>
```

Comment:
We *probably* do not need to raise every possible exception here, but; this function should be called from a process started with the very general `proc_lib:start{_link,monitor}/3,4,5`, which expects the started process to run to an end in any manner; any value or exception.

So if there is user code that today, for some reason throws some exception, say it wants the logging in the top level catch in `proc_lib:init_p/3,5`; to make it possible for that code to use the new `proc_lib:init_fail/2,3` and raise *any* exception, I think it should handle also the the really fishy in this context class `throw`.

Should a process unconditionally exit by, for example `process_flag(trap_exit, false), exit(self(), suicide)`, it will circumvent the top level catch in `init_p()`.  Should it catch the exception from `init_fail()` it would mess up the exception it itself requested.  Both cases, I think, are misuses that we cannot guard against.  The `proc_lib:start` process simply has to run to an end.